File: Sc.Models/Dtos/ReceivedVouchers/ReceivedVoucherCertificateDto.cs
using Sc.Models.Dtos.Suppliers;
using Sc.Models.Entities.Base;

namespace Sc.Models.Dtos.ReceivedVouchers
{
    public class ReceivedVoucherCertificateDto : Entity
    {
        public int ReceivedVoucherId { get; set; }

        public int UserId { get; set; }
        public string Username { get; set; }
        public string UserFullname { get; set; }

        public ReceivedVoucherCertificateFileDto File { get; set; }

        public int SupplierId { get; set; }
        public SupplierDto Supplier { get; set; }

        public int OfferingId { get; set; }
        public SupplierOfferingDto Offering { get; set; }
    }
}

File: WebApplication/src/received-vouchers/dtos/received-voucher-certificate.dto.ts
import { SupplierDto } from "src/suppliers/dtos/supplier.dto";
import { ReceivedVoucherCertificateFileDto } from "./received-voucher-certificate-file.dto";
import { SupplierOfferingDto } from "src/suppliers/dtos/supplier-offering.dto";

export class ReceivedVoucherCertificateDto {
    id: number;

    receivedVoucherId: number;

    userId: number;
    username: string;
    userFullname: string;

    file: ReceivedVoucherCertificateFileDto;

    supplierId: number;
    supplier: SupplierDto;

    offeringId: number;
    offering: SupplierOfferingDto;
}